app/src/main/java/com/internship/move/feature/map/MapFragment.kt

```diff
@@ -5,37 +5,50 @@ import android.view.View
 import androidx.fragment.app.Fragment
 import androidx.lifecycle.lifecycleScope
 import androidx.navigation.fragment.findNavController
-import com.internship.move.OnboardingViewModel
+import androidx.navigation.fragment.navArgs
 import com.internship.move.R
 import com.internship.move.databinding.FragmentMapBinding
+import com.internship.move.feature.authentication.AuthenticationViewModel
+import com.internship.move.feature.onboarding.OnboardingViewModel
+import com.internship.move.utils.logTag
 import com.zhuinden.fragmentviewbindingdelegatekt.viewBinding
-import kotlinx.coroutines.CoroutineScope
-import kotlinx.coroutines.Dispatchers
-import kotlinx.coroutines.SupervisorJob
 import kotlinx.coroutines.launch
 import org.koin.androidx.viewmodel.ext.android.viewModel
 
 class MapFragment : Fragment(R.layout.fragment_map) {
 
     private val binding by viewBinding(FragmentMapBinding::bind)
-    private val viewModel: OnboardingViewModel by viewModel()
+    private val onboardingViewModel: OnboardingViewModel by viewModel()
+    private val authViewModel: AuthenticationViewModel by viewModel()
+    private val args: MapFragmentArgs by navArgs()
 
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         super.onViewCreated(view, savedInstanceState)
 
         initListeners()
+        initViews()
     }
 
     private fun initListeners() {
         binding.logoutBtn.setOnClickListener {
+            logTag("LOGOOUT", authViewModel.userData.value.toString())
```

Comment:
This is the same bug as we discussed before: the lifecycle will get destroyed and all the calls will be cancelled.

Move it into the viewModel